packages/core/test/specs/data_sources/model/conditional_variables/ConditionalTraits.ts

```diff
@@ -0,0 +1,174 @@
+
+  it('should add a trait with a data-source condition', () => {
+    const dataSource: DataSourceProps = {
+      id: 'ds1',
+      records: [{ id: 'left_id', left: 'Name1' }],
+    };
+    dsm.add(dataSource);
+
+    const component = cmpRoot.append({
+      tagName: 'h1',
+      type: 'text',
+      traits: [
+        {
+          type: 'text',
+          name: 'value',
+          value: {
+            type: ConditionalVariableType,
+            condition: {
+              left: {
+                type: DataVariableType,
+                path: 'ds1.left_id.left',
+              },
+              operator: GenericOperation.equals,
+              right: 'Name1',
+            },
+            ifTrue: 'Valid name',
+            ifFalse: 'Invalid name',
+          },
+        },
+      ],
+    })[0];
+
+    const traitValue = component.getTrait('value').get('value');
+    expect(traitValue).toBe('Valid name');
+  });
+
+  it('should change trait value with changing data-source value', () => {
+    const dataSource: DataSourceProps = {
+      id: 'ds1',
+      records: [{ id: 'left_id', left: 'Name1' }],
+    };
+    dsm.add(dataSource);
+
+    const component = cmpRoot.append({
+      tagName: 'h1',
+      type: 'text',
+      traits: [
+        {
+          type: 'text',
+          name: 'value',
+          value: {
+            type: ConditionalVariableType,
+            condition: {
+              left: {
+                type: DataVariableType,
+                path: 'ds1.left_id.left',
+              },
+              operator: GenericOperation.equals,
+              right: 'Name1',
+            },
+            ifTrue: 'Correct name',
+            ifFalse: 'Incorrect name',
+          },
+        },
+      ],
+    })[0];
+
+    const traitValueBefore = component.getTrait('value').get('value');
+    expect(traitValueBefore).toBe('Correct name');
+
+    dsm.get('ds1').getRecord('left_id')?.set('left', 'Different name');
+    const traitValueAfter = component.getTrait('value').get('value');
+    expect(traitValueAfter).toBe('Incorrect name');
+  });
+
+  it('should throw an error if no condition is passed in trait', () => {
+    expect(() => {
+      cmpRoot.append({
+        tagName: 'h1',
+        type: 'text',
+        traits: [
+          {
+            type: 'text',
+            name: 'invalidTrait',
+            value: {
+              type: ConditionalVariableType,
+            },
+          },
+        ],
+      });
+    }).toThrow(MissingConditionError);
+  });
+
+  it('should store traits with conditional values correctly', () => {
+    const conditionalTrait = {
+      type: ConditionalVariableType,
+      condition: {
+        left: 0,
+        operator: NumberOperation.greaterThan,
+        right: -1,
+      },
+      ifTrue: 'Positive',
+    };
+    cmpRoot.append({
+      tagName: 'h1',
+      type: 'text',
+      traits: [
+        {
+          type: 'text',
+          name: 'dynamicTrait',
+          value: conditionalTrait,
```

Comment:
by default trait changes are added to the component as attributes, let's also ensure conditions are working with traits and `changeProp: true` (in that case trait changes are added to the component properties)